Hazel/src/Hazel/Debug/Instrumentor.h

```diff
@@ -107,8 +109,8 @@ namespace Hazel {
 		{
 			auto endTimepoint = std::chrono::high_resolution_clock::now();
 
-			long long start = std::chrono::time_point_cast<std::chrono::microseconds>(m_StartTimepoint).time_since_epoch().count();
-			long long end = std::chrono::time_point_cast<std::chrono::microseconds>(endTimepoint).time_since_epoch().count();
+			long double start = static_cast<long double>(std::chrono::time_point_cast<std::chrono::nanoseconds>(m_StartTimepoint).time_since_epoch().count()) / 1000;
```

Comment:
Should not you explicitly cast 1000 (integer) into long double ? Maybe try to compile with flags -Wconversion.
I'm a noob concerning chrono but can nanosecond may overflow for long durations? May you try to measure/profile a sleep(5 minutes) to see if this is ok. Can we imagine a case of profiling a game for one hour or more?

Reply:
Even for such a rare case, my solution would be to start a new session (from time 0) with a name like '.... (2)' or something when it overflows...

Reply:
For long durations (> minutes) I think you will do not care about nanoseconds ;) my idea would be:
- try casting from nanoseconds
- check if overflowed (if possible)
- try casting from milliseconds

But I'm not an expert of chrono.

Reply:
I watched CppCon 2016: Howard Hinnant “A chrono Tutorial" https://www.youtube.com/watch?v=P32hvk8b13M I understood now how this lib is working I think we can simplify code

Reply:
Got rid of the static_cast and /1000, using chrono to do the conversion now.  Also changed duration back to chrono:::microseconds since we only really care about the nanoseconds on the start time to prevent duplicates...